### ISSUE TITLE
Wrong return Type in 2.8.0

### ISSUE THREAD

**Issue by erenturkm** [vkhorikov/CSharpFunctionalExtensions#204]
Below is my Unit Test that started to fail on 2.8.0 but is working with 2.7.1. Basicly what I can see is Result.Success is not returning a version object but a Result.Version object.

```
[Fact] 
public void SuccessfullyConvertsVesionToToken() 
{ 
    bool T1 = false; 
    Version EI = new Version(0, 0, 0, 0); 
    Result<dynamic> gotRes = Token.ConvertToVersion(EI); 
    dynamic Res = gotRes.Value; 
    T1 = Res.GetType() == typeof(Version); 
   Assert.True(T1);
 } 

public static Result<dynamic> ConvertToVersion(dynamic V) 
{ 
    Version res7 = new Version(0, 0, 0, 0); 
    if (V.GetType() == typeof(Version)) return Result.Success(V); 
    if (V.GetType() == typeof(string) && Version.TryParse(V, out res7)) return Result.Success(res7); 
   else return Result.Failure<dynamic>("Could not convert to Version"); 
}

```

**Comment by vkhorikov** [comment]
Could also you post a version that worked with 2.7.1? This one doesn't seem to compile when I downgrade to 2.7.1.

**Comment by erenturkm** [comment]
It is my mistake, since return type differs, you have to cast it to dynamic when calling Result.Success. 2nd function should be (I am not clear on why formatting is wrong when I copy/paste):
`
 public static Result<dynamic> ConvertToVersion(dynamic V)
        {
            Version res7 = new Version(0, 0, 0, 0);
            if (V.GetType() == typeof(Version)) return Result.Success<dynamic>(V);
            if (V.GetType() == typeof(string) && Version.TryParse(V, out res7)) return Result.Success<dynamic>(res7);
            else return Result.Failure<dynamic>("Could not convert to Version");
        }
`
I just tested the code with both versions, it compiles and produces the following output. I also attached what I see in the debugger.
Version 2.8.0 Output:
![CSharpFuncExtensions2 8 0Output](https://user-images.githubusercontent.com/21189075/83229951-91ff9f00-a191-11ea-8743-96ecb3e6ee68.jpg)
Version 2.7.1 Output:
![CSharpFuncExtensions2 7 1Output](https://user-images.githubusercontent.com/21189075/83229955-9330cc00-a191-11ea-9733-3bab276b61e5.jpg)

**Comment by erenturkm** [comment]
Hi Vladimr,

Can you give any update on when we can expect a solution?

**Comment by vkhorikov** [comment]
This is fixed in 2.8.2
